components/content/magical-marbles/index.vue

```diff
@@ -0,0 +1,319 @@
+<script setup lang="ts">
+import { gsap } from 'gsap'
+import { Color, RepeatWrapping, NearestFilter, MeshStandardMaterial } from 'three'
+import { vertex, fragment } from './shaders'
+
+const gl = {
+    alpha: true,
+    shadows: false,
+}
+
+let ctx, tl
+
+const sphereRef = shallowRef(null)
+const backgroundRef = ref(null)
+const mainRef = ref(null)
+const indexColor = ref(0)
+const tlInProgress = ref(false)
+
+const colors = ref([
+    [0, 100, 50],
+    [60, 100, 50],
+    [150, 100, 50],
+    [240, 70, 60],
+    [0, 0, 80],
+])
+
+const params = reactive({
+    timeOffset: 0,
+    roughness: 0.15,
+    speed: 0.05,
+    iterations: 48,
+    depth: 0.6,
+    smoothing: 0.2,
+    displacement: 0.1,
+    metalness: 0,
+    colorB: "#008080",
+})
+
+const heightMap = await useTexture(['/magical-marbles/heightMap.jpeg'])
+const displacementMap = await useTexture(['/magical-marbles/displacementMap.jpeg'])
+heightMap.minFilter = displacementMap.minFilter = NearestFilter
+displacementMap.wrapS = displacementMap.wrapT = RepeatWrapping
+
+const { onLoop } = useRenderLoop()
+
+const { roughness, iterations, depth, smoothing, displacement, metalness, speed } = useControls({
```

Comment:
I can see you're using useControls here, but I can't see controls on the example this is on purpose?

Maybe you need to add the <TresLeches /> ¿?

Reply:
So why not adapt a custom url to display the TresLeches or not, for example via: `https://lab.tresjs.org/experiments/magical-marbles#debug`. 
This could be available for all pages? What do you think?

Reply:
unm, I believe this approach works well in some scenarios, in personal projects for example. But here, I am not sure. As this is a showcase people should be able to play with it.

In any case the final choice is up to you, but if you decide to add a #debug approach please provide some instructions so that people can access it.

Reply:
I agree with you, I'm going to leave TresLeches activated, there are some interesting values to change that could be fun! And in the future, why not add a drag'n'drop function on Leches to modify the height map in real time 😉

Reply:
Yes, Leches currently is stopped. We need to re-think the approach and we don't have enough hands for everything jejeje

Reply:
I can help on all packages if you want helps 😉 I'm starting slowly on the Lab haha!

Reply:
Thanks, is highly appreciated. If you have plans to contribute, please tell me and I'll add to the Tres Team. I'll wait for your PRs :D